src/StructureFactors.jl

```diff
@@ -145,43 +145,60 @@ function StructureFactor(snaps::Vector{Array{Vec3, 4}}, crystal; kwargs...)
     sf
 end
 
+"""
+Updates `M` in-place to hold the magnetization vectors obtained by scaling `s`
+ by the appropriate spin magnitudes and g-tensors in `sites_info`.
+This function assumes `M` has a first index of length 3, which correspond
+ to the magnetization components. (Rather than storing an Array{Vec3}).
+"""
+function _compute_mag!(M, sys::SpinSystem)
+    for b in 1:nbasis(sys)
+        Sg = sys.sites_info[b].S * sys.sites_info[b].g
+        for idx in eachcellindex(sys)
+            M[:, b, idx] .= Sg * sys[b, idx]
```

Comment:
Bohr magneton? Also I realize now that `S g` and `g' S` have different shapes (1x3 vs 3x1 matrices). It is probably best to stick with 3x1 matrices if we want to interpret `M` as a 3-vector.

Reply:
Same as above comments -- here `Sg` is the scalar spin magnitude times the g-tensor, so still a 3x3 matrix.

Reply:
I will change the names / multiplication order here, actually, just to reflect that the convention is indeed m = g S. That will prevent confusion in the future, thanks for catching this.

Reply:
Wait, I'm confused, the code as written `M[:, b, idx] .= Sg * sys[b, idx]` makes sense now, no need to change anything?

Reply:
I would just swap the code/names to be `g S` rather than `S g`. Since `S` here is a scalar, both ways are doing the same thing, it's just that this way will emphasize the convention `M = g S`.